modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache

```diff
@@ -70,60 +64,36 @@ export function {{classname}}FromJSONTyped(json: any, ignoreDiscriminator: boole
     {{#items}}
     {{#isDateType}}
     if (Array.isArray(json)) {
-        if (json.every(item => !(isNaN(new Date(json).getTime()))) {
-            return json.map(value => new Date(json);
+        if (json.every(item => !(isNaN(new Date(item).getTime())))) {
+            return json.map(item => new Date(item));
         }
     }
     {{/isDateType}}
     {{#isDateTimeType}}
     if (Array.isArray(json)) {
-        if (json.every(item => !(isNaN(new Date(json).getTime()))) {
-            return json.map(value => new Date(json);
+        if (json.every(item => !(isNaN(new Date(item).getTime())))) {
+            return json.map(item => new Date(item));
         }
     }
     {{/isDateTimeType}}
-    {{#isNumeric}}
-    if (Array.isArray(json)) {
-        if (json.every(item => typeof item === 'number'{{#isEnum}} && ({{#allowableValues}}{{#values}}item === {{.}}{{^-last}} || {{/-last}}{{/values}}{{/allowableValues}}){{/isEnum}})) {
-            return json;
-        }
-    }
-    {{/isNumeric}}
-    {{#isString}}
-    if (Array.isArray(json)) {
-        if (json.every(item => typeof item === 'string'{{#isEnum}} && ({{#allowableValues}}{{#values}}item === '{{.}}'{{^-last}} || {{/-last}}{{/values}}{{/allowableValues}}){{/isEnum}})) {
-            return json;
-        }
-    }
-    {{/isString}}
     {{/items}}
     {{/isArray}}
     {{^isArray}}
     {{#isDateType}}
     if (!(isNaN(new Date(json).getTime()))) {
-        return {{^required}}json == null ? undefined : {{/required}}({{#required}}{{#isNullable}}json == null ? null : {{/isNullable}}{{/required}}new Date(json));
```

Comment:
These null-checks are unnecessary because the function already returns in the beginning when `json` is null.